core/arch/arm/kernel/stmm_sp.c

```diff
@@ -716,20 +905,40 @@ static bool spm_handle_direct_req(struct thread_svc_regs *regs)
 
 static bool spm_handle_svc(struct thread_svc_regs *regs)
 {
-	switch (regs->x0) {
```

Comment:
Here's an attempt at avoid a duplicated `switch()` for the ARM32 case. I wonder if this wouldn't be simpler:
``` C
#ifdef ARM64
uint64_t *a0 = &regs->x0;
#define __SEND_DIRECT_RESP FFA_MSG_SEND_DIRECT_RESP_64
#define __SEND_DIRECT_REQ   FFA_MSG_SEND_DIRECT_REQ_64
#endif
#ifdef ARM32
uint32_t *a0 = &regs->x0;
#define __SEND_DIRECT_RESP FFA_MSG_SEND_DIRECT_RESP_32
#define __SEND_DIRECT_REQ   FFA_MSG_SEND_DIRECT_REQ_32
#endif
```
plus some some small adjustments furst down. I guess you get the picture.

Reply:
+1, thanks.